### PR TITLE
Updated _orderTableLength global to use gamestate_structure

### PR DESCRIPTION
**Pull request by memellis**

### CODE REVIEW

File: src/OpenLoco/src/GameCommands/Vehicles/CreateVehicle.cpp [hunk 1]
```diff
@@ -395,9 +394,9 @@ namespace OpenLoco::GameCommands
 
     static void sub_470312(VehicleHead* const newHead)
     {
-        _987C5C[_orderTableLength] = 0;
-        newHead->orderTableOffset = _orderTableLength;
-        _orderTableLength++;
+        _987C5C[OrderManager::numOrders()] = 0;
+        newHead->orderTableOffset = OrderManager::numOrders();
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/Vehicles/CreateVehicle.cpp:398]: Should really rename numOrders to orderTableLength but that's for another day.

Comment (memellis) [reply src/OpenLoco/src/GameCommands/Vehicles/CreateVehicle.cpp:398]: Happy to change numOrders to orderTableLength. Do you want that in a separate merge request?

Comment (AaronVanGeffen) [reply src/OpenLoco/src/GameCommands/Vehicles/CreateVehicle.cpp:398]: I think it's safe to do in this PR as well.

Comment (duncanspumpkin) [reply src/OpenLoco/src/GameCommands/Vehicles/CreateVehicle.cpp:398]: Yeah second pr

Comment (memellis) [reply src/OpenLoco/src/GameCommands/Vehicles/CreateVehicle.cpp:398]: OK I'll go for a second pull request.
Will this pull request get merged, or do I need to do the "change numOrders to orderTableLength" pull request first and then amend this pull request?

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**